packages/fast-components-react-base/src/image/image.tsx

```diff
@@ -0,0 +1,116 @@
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import Foundation, { HandledProps } from "../foundation";
+import { IImageHandledProps, IImageMangedClasses, IImageUnhandledProps } from "./image.props";
+import { IImageClassNameContract, IManagedClasses } from "@microsoft/fast-components-class-name-contracts";
+
+/**
+ * TODO: Hook viewports values into Styles
+ */
+export interface IBreakpoints {
+    vp1: number;
+    vp2: number;
+    vp3: number;
+    vp4: number;
+    vp5: number;
+    vp6: number;
+}
+
+const breakpoints: IBreakpoints = {
+    vp1: 0,
+    vp2: 540,
+    vp3: 768,
+    vp4: 1084,
+    vp5: 1400,
+    vp6: 1779
+};
+
+/* tslint:disable-next-line */
+class Image extends Foundation<IImageHandledProps & IManagedClasses<IImageClassNameContract>,  React.HTMLAttributes<HTMLImageElement | HTMLPictureElement>, {}> {
+    protected handledProps: HandledProps<IImageHandledProps & IManagedClasses<IImageClassNameContract>> = {
+        managedClasses: void 0,
+        alt: void(0),
+        itemScope: void(0),
+        round: void(0),
+        src: void(0),
+        vp1: void(0),
+        vp2: void(0),
+        vp3: void(0),
+        vp4: void(0),
+        vp5: void(0),
+        vp6: void(0),
+    };
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactElement<HTMLImageElement | HTMLPictureElement> {
+        if (this.props.src) {
+            return (
+                <img
+                    {...this.unhandledProps()}
+                    className={this.generateClassNames()}
+                    alt={this.props.alt}
+                    src={this.props.src}
+                />
+            );
+        } else {
+            const classes: string = this.generateRoundClass(this.props.managedClasses.image);
+
+            return (
+                <picture
+                    {...this.unhandledProps()}
+                    className={this.generateClassNames()}
+                >
+                    {this.generateSourceElement(6)}
+                    {this.generateSourceElement(5)}
+                    {this.generateSourceElement(4)}
+                    {this.generateSourceElement(3)}
+                    {this.generateSourceElement(2)}
+                    {this.generateSourceElement(1)}
+                    {this.props.vp1 ? <img srcSet={this.props.vp1}src={this.props.vp1} alt={this.props.alt} className={classes}/> : null}
+                </picture>
+            );
+        }
+    }
+
+    /**
+     * Generates class names using optional props
+     */
+    protected generateClassNames(): string {
+        let classNames: string = this.props.src ? this.props.managedClasses.image : this.props.managedClasses.picture;
+
+        if (this.props.src) {
```

Comment:
not sure `generateRoundClass` needs to be its own method, can we consolidate the logic in here and just check for `this.props.round`?

Reply:
We need to check if its a picture element or image first. Because borderRadius is not a property of picture

Reply:
Ah, I see now, looks good

Reply:
@eljefe223 & @janechu - I think we can combine these into one method without overcomplicating it. I think I had an example on a different PR somewhere. Using a boolean in the generate classNames function could say that this is the nested version of image which would apply only what is needed and then exit out before hitting what is needed for the root.

Reply:
I suggest we simplify here if we can.